File: snippets/go/args.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) == 2 {
		fmt.Printf("Hello, %s\n", os.Args[1])
	} else {
		fmt.Fprintf(os.Stderr, "Usage: %s <name>\n", os.Args[0])
		os.Exit(1)
	}
}

File: snippets/rs/args.rs
use std::env;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();

    match args.len() {
        2 => {
            println!("Hello, {}", args[1]);
            process::exit(0);
        }
        _ => {
            println!("Usage: {} <name>", args[0]);
            println!("Please provide exactly one argument for the name");
            process::exit(1);
        },
    }
}
